src/shared/api/MatchMinerAPI.ts

```diff
@@ -0,0 +1,51 @@
+import * as request from 'superagent';
+import { ITrial, ITrialMatch } from "shared/model/MatchMiner.ts";
+import { buildCBioPortalAPIUrl } from "./urls";
+
+/**
+ * Retrieves the trial matches for the query given, if they are in the MatchMiner API.
+ */
+
+const cbioportalUrl = buildCBioPortalAPIUrl('api-legacy/proxy/matchminer/api');
+export async function postMatchMinerTrialMatches(query: object): Promise<Array<ITrialMatch>> {
+    return request.post(cbioportalUrl + '/query_trial_match')
+    .set('Content-Type', 'application/json')
+    .send(query)
+    .then((res) => {
+        const response = JSON.parse(JSON.parse(res.text));
+        return response.map((record:any) => ({
+            nctId: record.nct_id,
+            oncotreePrimaryDiagnosisName: record.oncotree_primary_diagnosis_name,
+            gender: record.gender,
+            matchType: record.match_type,
+            armDescription: record.arm_description,
+            trueHugoSymbol: record.true_hugo_symbol,
+            trialAccrualStatus: record.trial_accrual_status,
+            matchLevel: record.match_level,
+            sampleId: record.sample_id,
+            mrn: record.mrn,
+            trueProteinChange: record.true_protein_change,
+            vitalStatus: record.vital_status,
+            genomicAlteration: record.genomic_alteration,
+            patientClinical: record.trial_age_numerical + ' ' + record.trial_oncotree_primary_diagnosis,
+            patientGenomic: record.true_hugo_symbol + ' ' + record.true_protein_change,
+            trialAgeNumerical: record.trial_age_numerical,
+            trialOncotreePrimaryDiagnosis: record.trial_oncotree_primary_diagnosis
+        }));
+    });
+}
+
+export async function getMatchMinerTrial(nctId: string): Promise<ITrial> {
```

Comment:
We also need to support a post method of this one. NCT ID is not always available, we need to send two parameters per trial. And we should support multi queries.

Reply:
For PatientViewPageStore.ts, I just created 3 functions(line 681 ~ 724).
```
readonly trialMatches = remoteData<Array<ITrialMatch>>({
        invoke: () => {
            return postMatchMinerTrialMatches({mrn: this.patientId});
        },
        onError: (err: Error) => {
            // fail silently
        }
    }, []);

    @computed get trialIds(): ITrialQuery {
        if (this.trialMatches.result) {
            let nctIds = new Set<string>();
            let protocolNos = new Set<string>();
            _.forEach(this.trialMatches.result, function(trialMatch: ITrialMatch) {
                if (_.isEmpty(trialMatch.protocolNo)) {
                    nctIds.add(trialMatch.nctId);
                } else {
                    protocolNos.add(trialMatch.protocolNo);
                }
            });
            return {
                nct_id: [...nctIds],
                protocol_no: [...protocolNos]
            };
        } else {
            return {
                nct_id: [],
                protocol_no: []
            };
        }
    }

    readonly matchMinerTrials = remoteData<Array<ITrial>>({
        invoke: async () => {
            if (this.trialIds.protocol_no.length > 0 || this.trialIds.nct_id.length > 0) {
                return postMatchMinerTrialsById(this.trialIds);
            } else {
                return [];
            }
        },
        onError: (err: Error) => {
            // fail silently
        }
    }, []);
```